### PR TITLE
Fix admission plugin registration

### LINKED ISSUE #50833: kube-apiserver admission control plugins registered too late
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

The current help message from kube-apiserver is wrong. It shows only the 'NamespaceLifecycle' plug-in is available, and the default value is `[AlwaysAdmit]`. Users cannot know what are the available plug-ins, and they get confused where `AlwaysAdmit` comes from.

```
kube-apiserver --help
Usage of ./kube-apiserver:
      --admission-control stringSlice                           Ordered list of plug-ins to do admission control of resources into cluster. Comma-delimited list of: NamespaceLifecycle. (default [AlwaysAdmit])
```

**What you expected to happen**:

The correct output should be this:

```
Usage of ./kube-apiserver:
      --admission-control stringSlice                           Ordered list of plug-ins to do admission control of resources into cluster. Comma-delimited list of: AlwaysAdmit, AlwaysDeny, AlwaysPullImages, DefaultStorageClass, DefaultTolerationSeconds, DenyEscalatingExec, DenyExecOnPrivileged, GenericAdmissionWebhook, ImagePolicyWebhook, InitialResources, Initializers, LimitPodHardAntiAffinityTopology, LimitRanger, NamespaceAutoProvision, NamespaceExists, NamespaceLifecycle, NodeRestriction, OwnerReferencesPermissionEnforcement, PersistentVolumeLabel, PodNodeSelector, PodPreset, PodSecurityPolicy, PodTolerationRestriction, Priority, ResourceQuota, SecurityContextDeny, ServiceAccount. (default [AlwaysAdmit])
...
```

**How to reproduce it (as minimally and precisely as possible)**:
Run `kube-apiserver --help` using master or at least 1.7.2.

**Anything else we need to know?**:

This is a regression introduced when refactoring code into generic server code.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.7.2+ ?
- Cloud provider or hardware configuration**: Any
- OS (e.g. from /etc/os-release): Any
- Kernel (e.g. `uname -a`): Any
- Install tools: Any
- Others: None

### PR DESCRIPTION
**Pull request by tengqm**
**What this PR does / why we need it**:
The current registration of admission plug-ins happen too late so the
command line help is not yielding correct/useful information to users.
This is already causing confusion as mentioned in some issues.

This patch fixes it by moving plugins.go down to options package because
"plugins" in this context are themselves options for users. Registration
of plugins is not an expensive operation and it is already done in most
execution paths.

**Which issue this PR fixes**
fixes #50833 

**Special notes for your reviewer**:
There might be other thoughts on how to better align this with other
efforts. This PR is leaving them to future work. For example:
in future, we may want to revisit these plugins and migrate them to the
shared apiserver repo when appropriate.

**Release note**:
I'm not sure if a release note is needed for this. Maybe there is a need.

```release-note
```

### CODE REVIEW

**Comment by dims** [comment]
@tengqm github shows 5 commits, can you squash them please?

**Comment by guangxuli** [comment]
@tengqm  before running `/retest`, you can check the bazel related scripts in you local.

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ const etcdRetryInterval = 1 * time.Second
 // NewAPIServerCommand creates a *cobra.Command object with default parameters
 func NewAPIServerCommand() *cobra.Command {
 	s := options.NewServerRunOptions()
+
```
Comment (dixudx) [inline cmd/kube-apiserver/app/server.go]: redundant newline ?

Comment (dixudx) [inline cmd/kube-apiserver/app/server.go]: @tengqm run `./hack/update-bazel.sh` in your local to update bazel.

Comment (tengqm) [inline cmd/kube-apiserver/app/server.go]: looks like the complaint now is not about bazel, it is something else.

**Pull request merged**